tools/monitoring/config_test.go

```diff
@@ -0,0 +1,228 @@
+      - job-name-regex: '.*'
+        occurrences: 50
+        jobs-affected: 20
+        prs-affected: 10 # for non-pull jobs, we don't care about the number of prs affected, so we set the number to 1, which will basically make this particular condition always true
+        period: 60`
+)
+
+var sampleConfig = Config{
+	Spec: []patternSpec{
+		{
+			ErrorPattern: "Something went wrong: starting e2e cluster: error creating cluster",
+			Hint:         "Check gcp status",
+			Alerts: []alertCondition{
+				{"^pull.*",
+					2,
+					1,
+					2,
+					1440},
+				{".*",
+					5,
+					2,
+					1,
+					1440},
+			},
+		},
+
+		{
+			"sample*error2",
+			"hint_for_pattern_2",
+			[]alertCondition{
+				{"^pull.*",
+					20,
+					10,
+					20,
+					60,
+				},
+				{".*",
+					50,
+					20,
+					10,
+					60,
+				},
+			},
+		},
+	},
+}
+
+func TestCollectErrorPatterns(t *testing.T) {
+	conf, err := newConfig([]byte(sampleYaml))
+	if err != nil {
+		t.Errorf("cannot parse yaml: %v", err)
+	}
+
+	type args struct {
+		f Config
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "CollectErrorPatterns",
+			args: args{
+				f: *conf,
+			},
+			want: []string{
+				"Something went wrong: starting e2e cluster: error creating cluster",
+				"sample*error2",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.args.f.CollectErrorPatterns(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("CollectErrorPatterns() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestConfig_Select(t *testing.T) {
+	type fields struct {
+		Spec []patternSpec
+	}
+	type args struct {
+		pattern string
+		jobName string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    SelectedConfig
+		wantErr bool
+	}{
+		{
+			name: "wild card pull job name pattern test",
+			fields: fields{
+				Spec: sampleConfig.Spec,
+			},
+			args: args{
+				pattern: "sample*error2",
+				jobName: "pull-is-me",
+			},
+			want: SelectedConfig{
+				Hint:         "hint_for_pattern_2",
+				Occurrences:  20,
+				JobsAffected: 10,
+				PrsAffected:  20,
+				Period:       60,
+			},
+		},
+
+		{
+			name: "wild card non-pull job name pattern test",
+			fields: fields{
+				Spec: sampleConfig.Spec,
+			},
+			args: args{
+				pattern: "sample*error2",
+				jobName: "not-pull-me",
+			},
+			want: SelectedConfig{
+				Hint:         "hint_for_pattern_2",
+				Occurrences:  50,
+				JobsAffected: 20,
+				PrsAffected:  10,
+				Period:       60,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f := Config{
+				Spec: tt.fields.Spec,
+			}
+			got, err := f.Select(tt.args.pattern, tt.args.jobName)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Config.Select() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Config.Select() = %v, want %v", got, tt.want)
```

Comment:
can you add more elaborate all the err messages? config.Select() does not mean much unless you know the code.